### PR TITLE
Pic 1163 - Track operation Id across SQS

### PR DESCRIPTION
**Pull request by jhevans**

### CODE REVIEW

**Comment by jhevans** [comment]
This is code complete but the corresponding implementation at the receiving end isn't yet so probably best to hold off merging until that has been successfully tested locally

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/service/SqsService.kt [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
+    @Autowired @Qualifier("getOperationId") private val getOperationId: () -> String
 ) {
 
     fun enqueueMessage(externalDocumentRequest: String): String {
         val msgRequest = SendMessageRequest()
             .withQueueUrl(getQueueUrl())
             .withMessageBody(externalDocumentRequest)
+            .withMessageAttributes(
+                mapOf(
+                    "operation_Id" to MessageAttributeValue()
```
Comment (chris-faulkner-mojdt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/service/SqsService.kt:26]: Can this just be `"operationId" to getOperationId()` ?
I think Strings are just the default

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/TelemetryConfig.kt [hunk 2]
```diff
@@ -40,4 +40,9 @@ class TelemetryConfig {
             .map { obj: RequestTelemetry -> obj.properties }
             .orElse(emptyMap())
     }
+
+    @Bean
+    fun getOperationId(): () -> String? {
```
Comment (chris-faulkner-mojdt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/TelemetryConfig.kt:45]: This is nullable which is OK but do we want to log or do something if that happens ?

**Review by chris-faulkner-mojdt — APPROVED** [review]
LGTM - couple of small points

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/TelemetryConfig.kt (hunk 2 above)
Comment (jhevans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/TelemetryConfig.kt:45]: Yeah good call, will update.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/service/SqsService.kt (hunk 1 above)
Comment (jhevans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/service/SqsService.kt:26]: The only method signature that's available as far as I can see requires type MessageAttributeValue - does seem a bit clunky I'll grant you but kinda makes sense in the context of the SQS payload where it maps directly like so:
```
"MessageAttributes": {
    "operation_Id": {
        "StringValue": "e2f47e6ceb8ead83ac1499c5e6f06bd0",
        "DataType": "String"
    }
}
```

**Pull request merged**